### PR TITLE
[Product AI v2] Remove feature flag and code cleanup

### LINKED ISSUE #12367: [Product AI v2] Remove the feature flag.

### LINKED ISSUE #12368: [Product AI v2] Remove code for old product creation with Ai flow

### PR DESCRIPTION
**Pull request by hichamboushaba**
Closes: #12367 Closes: #12368 

### Description
This PR removes the feature flag and the old unused code.

### Testing information
Test the happy path of the product creation AI feature and confirm it works without issues.

### Images/gif

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/232">20.1</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false"><p>This PR contains changes to Tracks-related logic. Please ensure (<strong>author and reviewer</strong>) the following are completed:</p>

<ul>
  <li>The tracks events must be validated in the Tracks system.</li>
  <li>Verify the internal Tracks spreadsheet has also been updated.</li>
  <li>Please consider registering any new events.</li>
  <li>The PR must be assigned the <strong>category: tracks</strong> label.</li>
</ul></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AIProductNameBottomSheet.kt [hunk 1]
```diff
@@ -1,411 +0,0 @@
...
-                        Icon(
-                            painter = painterResource(id = drawable.ic_ai),
-                            contentDescription = null, // decorative
-                            modifier = Modifier
-                                .padding(end = dimensionResource(id = dimen.minor_100))
-                                .size(dimensionResource(id = dimen.major_150)),
-                            tint = colorResource(id = color.woo_purple_60)
-                        )
-                        Text(
-                            text = stringResource(id = string.ai_product_name_sheet_title),
```
Comment (irfano) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AIProductNameBottomSheet.kt:121]: We can also remove
- `ai_product_name_sheet_title`
- `ai_product_name_sheet_subtitle`
- `ai_product_name_sheet_input_description`
- `ai_product_name_sheet_generate_button`
- `ai_product_name_sheet_regenerate_button`
- `ai_product_name_sheet_error_message`

from `strings.xml`.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AIProductNameBottomSheetFragment.kt [hunk 2]
```diff
@@ -1,70 +0,0 @@
...
-                is AIProductNameViewModel.CopyProductNameToClipboard -> copyProductNameToClipboard(
-                    event.productName
-                )
-            }
-        }
-    }
-
-    private fun copyProductNameToClipboard(name: String) {
-        try {
-            context?.copyToClipboard(getString(R.string.ai_product_name_copy_label), name)
```
Comment (irfano) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AIProductNameBottomSheetFragment.kt:63]: We can remove
- `ai_product_name_copy_label`
- `ai_product_name_copy_success`
- `ai_product_name_copy_error`

from `strings.xml`.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AIProductNameViewModel.kt [hunk 3]
```diff
@@ -1,157 +0,0 @@
...
-            }
-        )
-    }
-
-    private fun handleIdentificationSuccess(languageISOCode: String) {
-        identifiedLanguageISOCode = languageISOCode
-    }
-
-    private suspend fun generateProductName(languageISOCode: String) {
-        val result = aiRepository.generateProductName(
```
Comment (irfano) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AIProductNameViewModel.kt:57]: Now, there are unused functions in `AiRepository`. Can you also remove them?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AboutProductSubScreen.kt [hunk 4]
```diff
@@ -1,278 +0,0 @@
...
-                .padding(dimensionResource(id = R.dimen.major_100))
-        ) {
-            Column(
-                modifier = Modifier
-                    .weight(1f)
-                    .verticalScroll(rememberScrollState()),
-                verticalArrangement = Arrangement.spacedBy(dimensionResource(id = R.dimen.major_100)),
-            ) {
-                Text(
-                    text = stringResource(id = R.string.product_creation_ai_about_product_title),
```
Comment (irfano) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AboutProductSubScreen.kt:104]: We can also remove
- `product_creation_ai_about_product_title`
- `product_creation_ai_about_product_subtitle`
- `ai_product_creation_add_name_keywords_placeholder`
- `product_creation_ai_about_product_edit_text_caption`
- `product_creation_ai_about_product_set_tone`
- `product_creation_ai_about_product_continue_button`
- `product_creation_ai_tone_title`
- `product_creation_ai_tone_description`
- `product_creation_ai_tone_selected_content_desc`

from `strings.xml`.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/PackagePhotoBottomSheet.kt [hunk 5]
```diff
@@ -1,730 +0,0 @@
...
-                    }
-
-                    NoKeywordsFound -> {
-                        Message(stringResource(id = string.product_creation_package_photo_no_text_detected))
-                        Spacer(Modifier)
-                    }
-
-                    is Failure -> {
-                        Message(
-                            message = stringResource(id = string.product_creation_package_photo_error),
```
Comment (irfano) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/PackagePhotoBottomSheet.kt:174]: We can also remove
- `product_creation_package_photo_error`
- `product_creation_ai_preview_description_section`
 
from `strings.xml`.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/PackagePhotoViewModel.kt [hunk 6]
```diff
@@ -1,264 +0,0 @@
...
-
-    private fun analyzePackagePhoto() {
-        _viewState.update { _viewState.value.copy(state = Scanning) }
-        launch {
-            textRecognitionEngine.processImage(_viewState.value.imageUrl)
-                .onSuccess { keywords ->
-                    tracker.track(
-                        AnalyticsEvent.ADD_PRODUCT_FROM_IMAGE_SCAN_COMPLETED,
-                        mapOf(
-                            AnalyticsTracker.KEY_SCANNED_TEXT_COUNT to keywords.size
```
Comment (irfano) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/PackagePhotoViewModel.kt:62]: We can remove these keys:
image
image

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/ProductNameSubScreen.kt [hunk 7]
```diff
@@ -1,244 +0,0 @@
...
-            .fillMaxSize()
-    ) {
-        Column(
-            modifier = Modifier
-                .weight(1f)
-                .verticalScroll(rememberScrollState())
-        ) {
-            Spacer(modifier = Modifier.height(dimensionResource(id = R.dimen.major_100)))
-            Text(
-                text = stringResource(id = R.string.ai_product_creation_add_name_title),
```
Comment (irfano) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/ProductNameSubScreen.kt:105]: We can also remove
- `ai_product_creation_add_name_title`
- `ai_product_creation_add_name_subtitle`
- `ai_product_creation_add_name_keywords_label`
- `ai_product_creation_add_name_suggest_name_button`

from `strings.xml`.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/ProductPreviewSubScreen.kt [hunk 8]
```diff
@@ -1,408 +0,0 @@
...
-            .padding(dimensionResource(id = R.dimen.major_100))
-    ) {
-        Spacer(modifier = Modifier.height(dimensionResource(id = R.dimen.major_100)))
-        Text(
-            text = stringResource(id = R.string.product_creation_ai_preview_title),
-            style = MaterialTheme.typography.h5
-        )
-        Spacer(modifier = Modifier.height(dimensionResource(id = R.dimen.major_100)))
-        Text(
-            text = stringResource(id = R.string.product_creation_ai_preview_subtitle_legacy),
```
Comment (irfano) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/ProductPreviewSubScreen.kt:75]: We can also remove 
- `product_creation_ai_preview_subtitle_legacy`

from `strings.xml`.

**Review by irfano — CHANGES_REQUESTED** [review]
Thanks for cleaning up the repo. I found more code that can be tidied up. We could use AS's command to find unused string resources, but I believe I've already listed all the new unused strings that this PR introduced.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AIProductNameBottomSheet.kt (hunk 1 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AIProductNameBottomSheet.kt:121]: Done in 278661b371468438169a404224670c8e047d627e

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AIProductNameBottomSheetFragment.kt (hunk 2 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AIProductNameBottomSheetFragment.kt:63]: Done in 278661b371468438169a404224670c8e047d627e

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AboutProductSubScreen.kt (hunk 4 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AboutProductSubScreen.kt:104]: Done in 4df2243e4920b2eae24c9ca96fd159c41ecbe531

`product_creation_ai_tone_title` is still used though, so I kept it.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/PackagePhotoBottomSheet.kt (hunk 5 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/PackagePhotoBottomSheet.kt:174]: Done in 4df2243e4920b2eae24c9ca96fd159c41ecbe531

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/ProductNameSubScreen.kt (hunk 7 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/ProductNameSubScreen.kt:105]: Done in 278661b371468438169a404224670c8e047d627e

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/ProductPreviewSubScreen.kt (hunk 8 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/ProductPreviewSubScreen.kt:75]: Done in 278661b371468438169a404224670c8e047d627e

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/PackagePhotoViewModel.kt (hunk 6 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/PackagePhotoViewModel.kt:62]: Done in 980e6436553ce115b29af1998012360a317488b1

`VALUE_PRODUCT_CREATION` is still used, so it's kept.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AIProductNameViewModel.kt (hunk 3 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AIProductNameViewModel.kt:57]: Done in 4f3460fb0c0112c2101dd0286506f59b071905f6

**Comment by hichamboushaba** [comment]
Thanks @irfano for the great feedback, this is ready for another round.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AIProductNameViewModel.kt (hunk 3 above)
Comment (irfano) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/AIProductNameViewModel.kt:57]: You didn't remove the `AiRepository.generateProductLegacy()` function. Is there a reason to keep it?

**Review by irfano — CHANGES_REQUESTED** [review]
There is one more function in `AiRepository` that I think was overlooked. We’re good to merge once it is removed. Thank you for addressing all my comments. 👍🏻

**Comment by hichamboushaba** [comment]
Sorry for missing this function, it is removed now

**Review by irfano — APPROVED** [review]
LGTM! 👍🏻

**Pull request merged**